agent/checks/grpc.go

```diff
@@ -49,11 +50,23 @@ func NewGrpcHealthProbe(target string, timeout time.Duration, tlsConfig *tls.Con
 
 // Check if the target of this GrpcHealthProbe is healthy
 // If nil is returned, target is healthy, otherwise target is not healthy
-func (probe *GrpcHealthProbe) Check(target string) error {
+func (probe *GrpcHealthProbe) Check(target string, header map[string][]string) error {
 	serverAndService := strings.SplitN(target, "/", 2)
 	serverWithScheme := fmt.Sprintf("%s:///%s", resolver.GetDefaultScheme(), serverAndService[0])
 
 	ctx, cancel := context.WithTimeout(context.Background(), probe.timeout)
+	if len(header) > 0 {
+		var headerMD map[string]string = map[string]string{}
+		var kv []string
+
+		for k, v := range header {
+			headerMD[k] = strings.Join(v, ", ")
+			kv = append(kv, k)
+			kv = append(kv, headerMD[k])
```

Comment:
nit: I don't think we need `headerMD`. How about change it to `kv = append(kv, k, strings.Join(v, ", "))`